tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -368,7 +368,18 @@ static void test_nanocoap__get_multi_query(void)
     coap_get_uri_query_string(&pkt, query, sizeof(query));
     /* skip initial '&' from coap_get_uri_query_string() */
     TEST_ASSERT_EQUAL_STRING((char *)qs, &query[1]);
+
+    const char *val;
+    size_t val_len;
+    TEST_ASSERT(coap_find_uri_query(&pkt, "ab", &val, &val_len));
+    TEST_ASSERT_EQUAL_INT(3, val_len);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(val, "cde", val_len));
+    TEST_ASSERT(coap_find_uri_query(&pkt, "f", &val, &val_len));
+    TEST_ASSERT_EQUAL_INT(0, val_len);
+    TEST_ASSERT_EQUAL_INT((uintptr_t)NULL, (uintptr_t)val);
+    TEST_ASSERT(!coap_find_uri_query(&pkt, "cde", &val, &val_len));
 }
```

Comment:
As it is documented behavior, maybe add a test that checks for correct handling when passing NULL for val and val_len.